homeassistant/components/konnected/switch.py

```diff
@@ -41,7 +41,8 @@ def __init__(self, device_id, pin_num, data):
         self._pause = self._data.get(CONF_PAUSE)
         self._repeat = self._data.get(CONF_REPEAT)
         self._state = self._boolean_state(self._data.get(ATTR_STATE))
-        self._unique_id = '{}-{}'.format(device_id, PIN_TO_ZONE[pin_num])
+        self._unique_id = '{}-{}'.format(device_id,
+                                         hash(frozenset(data.items())))
```

Comment:
Since this includes config items, any change to the switch config will make the hash change, right? That's not optimal.

https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements

Reply:
In this case what we want is a unique ID based off of the `device_id` (MAC addr), pin number, and the config variables. The use case here is that users can create multiple entities that behave differently on the same physical switch. For example, a piezo buzzer can be configured as a door chime (three short beeps) _and_ as a warning alarm (infinitely repeating beeps).

The previous implementation was using only the device_id and pin, which failed in this case. So, including the config options in the hash would make each entity unique.

Does that make sense? I'm open to other suggestions.

Reply:
The suggestion fills the first requirement of unique_id, ie uniqueness. It doesn't fill the second requirement on stability.

Can we find something else to distinguish entitirs, that isn't user configurable?

Reply:
In this particular case (where the same physical switch/pin is used as multiple entities) there is nothing else besides user configuration to distinguish the entities.

Reply:
We should at least condense the config items that are used to compute the hash for the unique_id to the items that are related to the function of the entity.

Eg name is not related to the function of the entity.

Reply:
Yep, you're right. I just had the same idea. Updated.

Reply:
I had configured my switches to perform differently according to name 

```
      switches: 
        - zone: out
          name: Arm Away
          momentary: 1000
        - zone: out
          name: Arm Stay
          momentary: 3000
        - zone: out
          name: Disarm
          momentary: 1000
```
After upgrade I only have the "Disarm" switch showing as an entity.   Thoughts on how to fix?

Reply:
Please open an issue if you suspect a bug. If you need help please use our help channels:
https://home-assistant.io/help/#communication-channels

Merged PRs should not be used for support or bug reports. Thanks!